### ISSUE TITLE
Embbeding layer

### ISSUE THREAD

**Issue by wangyf8848** [FoundationVision/LlamaGen#66]
May I ask why VQVAE's learned codebook was not used as the Embedding layer for autoregressive training of language models, 
but only the indices of the codebook were used, and a new learnable Embedding layer was created for GPT as bellow：
`
self.tok_embeddings = nn.Embedding(config.vocab_size, config.dim)
`

**Comment by QuLiao1117** [comment]
> May I ask why VQVAE's learned codebook was not used as the Embedding layer for autoregressive training of language models, but only the indices of the codebook were used, and a new learnable Embedding layer was created for GPT as bellow： `self.tok_embeddings = nn.Embedding(config.vocab_size, config.dim)`

Any idea on this design choice? Would this initialization better than linear proj from learned codebook?
